core/controllers/suggestion.py

```diff
@@ -250,22 +538,37 @@ def put(self, target_id, suggestion_id):
                 suggestion.target_type, suggestion.target_id,
                 feconf.IMAGE_CONTEXT_QUESTION_SUGGESTIONS, suggestion.target_id,
                 target_image_filenames)
-        elif action == constants.ACTION_REJECT_SUGGESTION:
-            suggestion_services.reject_suggestion(
-                suggestion_id, self.user_id, self.payload.get('review_message'))
         else:
-            raise self.InvalidInputException('Invalid action.')
+            assert action == constants.ACTION_REJECT_SUGGESTION
+            suggestion_services.reject_suggestion(
+                suggestion_id, self.user_id,
+                self.normalized_payload['review_message']
+            )
 
         self.render_json(self.values)
 
 
-class SuggestionsProviderHandler(base.BaseHandler):
+class SuggestionsProviderHandler(
+    base.BaseHandler[
+        _NormalizedPayloadDictType,
+        _NormalizedRequestDictType
+    ]
+):
     """Provides suggestions for a user and given suggestion type."""
 
     GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    # Here we use type Any because the sub-classes of SuggestionsProviderHandler
```

Comment:
We shouldn't use Any. There are only a small fixed number of possible schemas so this should be a union or similar. Ditto below.

Reply:
I think no because even if there are small fixed number of schemas, we cannot remove Any from here because those schemas are the complex dicts and defining type for them is going to use a union with a very deep nesting of types, and maintaining a union of all those types is an ugly thing.

Reply:
But you still can't use Any. I think the issue is actually that in this case there are a small fixed number of schemas. You can name the subtypes so that you refer to them by their names rather than the complex dicts -- this is basically the same thing as all the other cases we've been tackling with change cmds etc.

As usual, union is OK if at this point in the code, all you know is that it's a union. When the code specializes, you can cast to a more specific type (with reasoning). But we can't be using Any here, it's as good as not typing at all.

Reply:
Added type, and removed Any!